index.js

```diff
@@ -148,9 +163,9 @@ Tokens.prototype.verify = function verify (secret, token) {
  * @private
  */
 
-function hash (str) {
+Tokens.prototype.hash = function hash (str) {
```

Comment:
this is the biggest change: moving this method from a private "helper" to a public class method, all necessitated by needing access to `this.hashAlgorithm`.

Is the naming convention to prepend an underscore to indicate private? (which would be great, IMHO -- `_hash`)